packages/emotion-primitives/src/createEmotion.js

```diff
@@ -0,0 +1,111 @@
+import * as React from 'react'
+import reactPrimitives from 'react-primitives'
+import {
+  channel,
+  contextTypes,
+  setTheme,
+  testAlwaysTrue,
+  pickAssign
+} from './utils'
+
+import { getStyles } from './getStyles'
+import { convertToRNStyles } from './convertToRNStyles'
+
+const primitives = ['Text', 'View', 'Image']
+
+const isValidPrimitive = primitive => primitives.indexOf(primitive) > -1
+
+// Validate and return the primitive
+const getPrimitive = primitive => {
+  if (typeof primitive === 'string') {
+    if (isValidPrimitive(primitive)) {
+      return reactPrimitives[primitive]
+    }
+    throw new Error(
+      `Cannot style invalid primitive ${primitive}. Expected primitive to be one of ['Text', 'View', 'Image']`
+    )
+  }
+
+  return primitive
```

Comment:
I've also removed `typeof primitive === 'function'` check because it would result in undefined being passed to `createElement`, we still might pass invalid value, but at least react might be able to report the invalid one (that it got) instead of reporting that it got `"undefined"`

Reply:
That's better!